package/android/cpp/rnskia-android/SkiaOpenGLSurfaceFactory.cpp

```diff
@@ -0,0 +1,220 @@
+#include <SkiaOpenGLSurfaceFactory.h>
+
+namespace RNSkia {
+/**
+ * The first context created will be considered the parent / shared context and
+ * will be used as the parent / shareable context when creating subsequent
+ * contexts.
+ */
+EGLContext BaseSkiaSurfaceFactory::_SharedEglContext = EGL_NO_CONTEXT;
+
+thread_local SurfaceFactoryContext WindowedSurfaceFactory::_skiaOpenGlContext;
+
+BaseSkiaSurfaceFactory::BaseSkiaSurfaceFactory(SkiaSurfaceType type, int width,
+                                               int height) {
+  _glSurface = EGL_NO_SURFACE;
+  _type = type;
+  _width = width;
+  _height = height;
+}
+
+bool BaseSkiaSurfaceFactory::resize(int width, int height) {
+  _width = width;
+  _height = height;
+
+  return true;
+}
+
+bool BaseSkiaSurfaceFactory::makeCurrent() {
+  if (!eglMakeCurrent(getContext()->glDisplay, _glSurface, _glSurface,
+                      getContext()->glContext)) {
+    RNSkLogger::logToConsole("eglMakeCurrent failed: %d\n", eglGetError());
+    return false;
+  }
+  return true;
+}
+
+bool BaseSkiaSurfaceFactory::initializeOpenGL(SurfaceFactoryContext *context) {
+  // Get default display
+  context->glDisplay = eglGetDisplay(EGL_DEFAULT_DISPLAY);
+  if (context->glDisplay == EGL_NO_DISPLAY) {
+    RNSkLogger::logToConsole("eglGetDisplay failed : %i", glGetError());
+    return false;
+  }
+
+  EGLint major;
+  EGLint minor;
+
+  if (!eglInitialize(context->glDisplay, &major, &minor)) {
+    RNSkLogger::logToConsole("eglInitialize failed : %i", glGetError());
+    eglTerminate(context->glDisplay);
+    context->glDisplay = EGL_NO_DISPLAY;
+    return false;
+  }
+
+  return true;
+}
+
+EGLConfig BaseSkiaSurfaceFactory::getConfig(EGLDisplay glDisplay) {
+
+  EGLint att[] = {EGL_RENDERABLE_TYPE,
+                  EGL_OPENGL_ES2_BIT,
+                  EGL_ALPHA_SIZE,
+                  8,
+                  EGL_BLUE_SIZE,
+                  8,
+                  EGL_GREEN_SIZE,
+                  8,
+                  EGL_RED_SIZE,
+                  8,
+                  EGL_DEPTH_SIZE,
+                  0,
+                  EGL_STENCIL_SIZE,
+                  0,
+                  EGL_NONE};
+
+  EGLint numConfigs;
+  EGLConfig glConfig = 0;
+  if (!eglChooseConfig(glDisplay, att, &glConfig, 1, &numConfigs) ||
+      numConfigs == 0) {
+    RNSkLogger::logToConsole(
+        "Failed to choose a config for %s surface. Error code: %d\n",
+        eglGetError());
+    return 0;
+  }
+
+  return glConfig;
+}
+
+EGLContext BaseSkiaSurfaceFactory::createOpenGLContext(EGLDisplay glDisplay) {
+  // Create OpenGL context
+  EGLint contextAttribs[] = {EGL_CONTEXT_CLIENT_VERSION, 2, EGL_NONE};
+
+  // Initialize the offscreen context for this thread
+  EGLContext newGLContext = eglCreateContext(glDisplay, getConfig(glDisplay),
+                                             _SharedEglContext, contextAttribs);
```

Comment:
this is the line that breaks the symmetry between Skia/OpenGL because  the OpenGL context has a parent but the Skia context doesn't